server/coordinator/scheduler.go

```diff
@@ -83,27 +84,31 @@ func (s *Scheduler) checkNode(ctx context.Context, ticker *time.Ticker) {
 				log.Error("get node shards failed", zap.Error(err))
 				continue
 			}
-			nodeShardsMapping := map[string][]*cluster.ShardInfo{}
+			nodeShardsMapping := map[string][]cluster.ShardInfo{}
 			for _, nodeShard := range nodeShards.NodeShards {
-				_, exists := nodeShardsMapping[nodeShard.Endpoint]
+				_, exists := nodeShardsMapping[nodeShard.ShardNode.Node]
 				if !exists {
-					nodeShardsMapping[nodeShard.Endpoint] = []*cluster.ShardInfo{}
+					nodeShardsMapping[nodeShard.ShardNode.Node] = []cluster.ShardInfo{}
 				}
-				nodeShardsMapping[nodeShard.Endpoint] = append(nodeShardsMapping[nodeShard.Endpoint], nodeShard.ShardInfo)
+				nodeShardsMapping[nodeShard.ShardNode.Node] = append(nodeShardsMapping[nodeShard.ShardNode.Node], cluster.ShardInfo{
+					ID:      nodeShard.ShardNode.ID,
+					Role:    nodeShard.ShardNode.ShardRole,
+					Version: nodeShards.ClusterTopologyVersion,
+				})
 			}
 			s.processNodes(ctx, nodes, t, nodeShardsMapping)
 		}
 	}
 }
 
-func (s *Scheduler) processNodes(ctx context.Context, nodes []*cluster.RegisteredNode, t time.Time, nodeShardsMapping map[string][]*cluster.ShardInfo) {
+func (s *Scheduler) processNodes(ctx context.Context, nodes []cluster.RegisteredNode, t time.Time, nodeShardsMapping map[string][]cluster.ShardInfo) {
 	for _, node := range nodes {
 		// Determines whether node is expired.
 		if !node.IsExpired(uint64(t.Unix()), heartbeatKeepAliveIntervalSec) {
 			// Shard versions of CeresDB and CeresMeta may be inconsistent. And close extra shards and open missing shards if so.
 			realShards := node.GetShardInfos()
-			expectShards := nodeShardsMapping[node.GetMeta().GetName()]
-			err := s.applyMetadataShardInfo(ctx, node.GetMeta().GetName(), realShards, expectShards)
+			expectShards := nodeShardsMapping[node.GetNode().Name]
```

Comment:
```suggestion
			expectShards := nodeShardsMapping[node.GetNode().NodeName]
```

Reply:
The structure is `Node`, and the name in `Node` do not need Node prefix.